### DISCUSSION TITLE
Is Node required?

### DISCUSSION THREAD

**Discussion by mnapoli** [spatie/shiki-php#4] [Q&A] [answered]
The README isn't really clear as to how Shiki runs behind the scenes.

Do we need Node? Is this package starting a subprocess?

Does that also mean we need to deploy `node_modules` with the PHP application?

Thanks

**Comment by riasvdv** [comment, marked as answer]
Yes you need Node as it runs a subprocess. The readme clearly states the following:

![image](https://user-images.githubusercontent.com/3626559/125455751-3aea9ece-6aff-4d64-8ad0-b033e89a230d.png)

**Reply by mnapoli** [reply]
Ah how did I miss that 🤦  

Thank you, so I guess then it's necessary to deploy `node_modules` too. Thanks for answering!

**Comment by aarondfrancis** [comment]
@mnapoli if you're not keen on deploying the entirety of your `node_modules`, you can check out https://torchlight.dev, no JavaScript required. Early access will go out at the end of this week.

**Reply by mnapoli** [reply]
I did. Quick feedback: it was very hard to be 100% sure how that works. Took me 5 minutes and digging into the source to understand that it's an HTTP API, that you have to sign up with tokens.

I know the landing page mentions "API", but that term can mean anything.

And finally I don't think I would use it for server-side rendering because of the latency and failure possibilities it could introduce in my application.

Hope that feedback is useful!

**Reply by aarondfrancis** [reply]
It is, thanks! There's definitely a tradeoff between hosting something yourself and having someone else host it. 

If you find Shiki prohibitively slow, gimme a shout 😉 

Best of luck!
